src/modules/aws_sns/mappers/topic.ts

```diff
@@ -0,0 +1,258 @@
+      isEqual(a.signatureVersion, b.signatureVersion) &&
+      isEqual(a.tracingConfig, b.tracingConfig) &&
+      isEqual(a.dataProtectionPolicy, b.dataProtectionPolicy)
+    );
+  };
+
+  getTopicAttributes = crudBuilderFormat<SNS, 'getTopicAttributes', Record<string, string> | undefined>(
+    'getTopicAttributes',
+    arn => ({ TopicArn: arn }),
+    res => res?.Attributes,
+  );
+
+  getTopicDataProtection = crudBuilderFormat<SNS, 'getDataProtectionPolicy', string | undefined>(
+    'getDataProtectionPolicy',
+    arn => ({ ResourceArn: arn }),
+    res => res?.DataProtectionPolicy,
+  );
+
+  createTopic = crudBuilderFormat<SNS, 'createTopic', string | undefined>(
+    'createTopic',
+    input => input,
+    res => res?.TopicArn,
+  );
+
+  getTopics = paginateBuilder<SNS>(paginateListTopics, 'Topics', undefined, undefined);
+
+  setTopicAttributes = crudBuilder2<SNS, 'setTopicAttributes'>('setTopicAttributes', input => input);
+
+  putDataProtectionPolicy = crudBuilder2<SNS, 'putDataProtectionPolicy'>(
+    'putDataProtectionPolicy',
+    input => input,
+  );
+
+  deleteTopic = crudBuilder2<SNS, 'deleteTopic'>('deleteTopic', TopicArn => ({
+    TopicArn,
+  }));
+
+  attributeKeys = [
+    'contentBasedDeduplication',
+    'deliveryPolicy',
+    'displayName',
+    'kmsMasterKeyId',
+    'policy',
+    'signatureVersion',
+    'tracingConfig',
+  ];
+
+  capitalize = (s: string) => {
+    return s.charAt(0).toUpperCase() + s.slice(1);
+  };
+
+  async topicMapper(arn: string, region: string, ctx: Context) {
+    if (!arn) return undefined;
+    let out = new Topic();
+
+    // if we have the topic, we can query the attributes and data protection
+    const client = (await ctx.getAwsClient(region)) as AWS;
+    const attributes = await this.getTopicAttributes(client.snsClient, arn);
+    if (attributes) {
+      for (const key of this.attributeKeys) {
+        const transformedKey = this.capitalize(key);
+        if (attributes[transformedKey]) {
+          const newValues = {
+            [key]: attributes[transformedKey],
+          };
+          out = Object.assign(out, newValues);
+        } else {
+          const newValues = {
+            [key]: undefined,
+          };
+          out = Object.assign(out, newValues);
+        }
+      }
+      out.fifoTopic = attributes.FifoTopic === 'true';
+    }
+
+    // if not fifo, get data protection
+    out.dataProtectionPolicy = undefined;
+    if (!out.fifoTopic) {
+      const dataProtection = await this.getTopicDataProtection(client.snsClient, arn);
+      if (dataProtection) out.dataProtectionPolicy = dataProtection;
+    }
+
+    out.arn = arn;
+    out.region = region;
+    out.name = parseArn(arn).resource;
+    return out;
+  }
+
+  cloud: Crud2<Topic> = new Crud2({
+    create: async (es: Topic[], ctx: Context) => {
+      const out = [];
+      for (const e of es) {
+        if (!e.name || !e.region) continue; // cannot create topic without name or region
```

Comment:
`name` and `region` are both non-nullable. I think we have the extra check for `region` and it has to be a valid region (due to FK). So maybe this is redundant.